### DISCUSSION TITLE
Is it possible to solve a shock-tube problem using Modelica.Fluid.Pipes.DynamicPipe

### DISCUSSION THREAD

**Discussion by octave-user** [OpenModelica/OpenModelica#12738] [Q&A]
Dear OpenModelica community,

Is there a way to solve a so called ["shock-tube problems"](https://www.astro.uu.se/~bf/course/numhd_course/Contents.html) using OpenModelica's DynamicPipe? In general, this is a typical test case for Roe solvers.

I'm attaching a naive implementation of the shock-tube problem:
[shock_tube.zip](https://github.com/user-attachments/files/16325664/shock_tube.zip)

Do you think that there is a way to obtain reasonable results from such model, or is it something which is not recommended even if an appropriate CFL number is used?

Reinhard

**Comment by casella** [comment]
Modelica.Fluid is not meant to solve gasdynamics problems. In many places, the kinetic term is neglected to avoid unecessarily difficult problems at low Mach numbers. 

You can check [this paper](http://dx.doi.org/10.3384/ecp1207681) on the topic.

**Reply by octave-user** [reply]
OK, but how can I interpret this comment in the source code of DynamicPipe?
`
model DynamicPipe "Dynamic pipe model with storage of mass and energy"
`
Does it mean that the term "energy" is referring just to internal energy, not kinetic energy?

**Reply by casella** [reply]
Yes. Enthalpy, in fact, as it is appropriate for systems with mass flow.

**Reply by casella** [reply]
Modelica.Fluid is already trying to (optionally) accommodate many different phenomena. Some said too many, and mentioned the German mythical Eierlegende Wollmilchsau. If you also add kinetic energy, you also get silk on top of eggs, milk etc. Probably too much 😅

**Reply by octave-user** [reply]
OK, it is clear now.

**Reply by octave-user** [reply]
> Modelica.Fluid is not meant to solve gasdynamics problems. In many places, the kinetic term is neglected to avoid unecessarily difficult problems at low Mach numbers.
> 
> You can check [this paper](http://dx.doi.org/10.3384/ecp1207681) on the topic.

Regarding the approach presented in the paper: [ecp1207681](http://dx.doi.org/10.3384/ecp1207681)
If I'm looking at the definitions of the connector presented in the paper, it seems to be completely incompatible with the standard connector from the Modelica Fluid library.
```
connector Stencil_a
"Interface for quasi one-dimensional high-speed flow"
```
```
connector FluidPort
  "Interface for quasi one-dimensional fluid flow in a piping network (incompressible or compressible, one or more phases, one or more substances)"
```
In the end, it was not my intent to solve a shock-tube problem. This was just a simple test case in order to check if the DynamicPipe would be usable for a more complex scenario. But if the connectors which were used in the paper are not compatible with the Modelica Fluid library, I fear that it would be necessary to implement everything from scratch. In addition to that, it seems that the code in the paper is using a kind of high order upwind interpolation. Probably this kind of approach would make it even more difficult to handle things like discontinuities of the cross section in a pipe or tee-junctions.

**Comment by octave-user** [comment]
On the other hand, the approach for a dynamic pipe considering fluid inertia, which is presented here [MathWorks Pipe](https://ch.mathworks.com/help/hydro/ref/pipetl.html#responsive_offcanvas), seems to be quite straight-forward, at least from the programming point of view.

Do you think, that this kind of momentum balance could be implemented somehow as an enhancement for the DynamicPipe in the Modelica Fluid library, or is it something which is not recommended?
![image](https://github.com/user-attachments/assets/0cac7629-a569-46de-898d-e5945f7a0af6)
